src/ORM/Rule/ExistsIn.php

```diff
@@ -40,14 +40,28 @@ class ExistsIn
     protected $_repository;
 
     /**
+     * Options for the constructor
+     *
+     * @var array
+     */
+    protected $_options = [];
+
+    /**
      * Constructor.
      *
+     * Available option for $options is 'partialNullsPass' flag. 
+     * Set to true to allow passing of composite foreign keys where one or multiple keys are null.
+     *
      * @param string|array $fields The field or fields to check existence as primary key.
      * @param object|string $repository The repository where the field will be looked for,
      * or the association name for the repository.
+     * @param array $options The options that modify the rules behavior.
      */
-    public function __construct($fields, $repository)
+    public function __construct($fields, $repository, array $options = [])
     {
+        $options += ['partialNullsPass' => false];
```

Comment:
I'm wondering if we've found a good name for this:

* `acceptNulls`
* `allowNullValues`